bldenv/man/QueryStatus.man
# @(#)37	1.3  src/bldenv/man/QueryStatus.man, bldprocess, bos412, GOLDA411a 6/1/92 12:32:19
#
#
# COMPONENT_NAME: (BLDTOOLS) BAI Build Tools
#
# FUNCTIONS:  QueryStatus man page
#
# ORIGINS: 27
#
# (C) COPYRIGHT International Business Machines Corp. 1991
# All Rights Reserved
# Licensed Materials - Property of IBM
#
# US Government Users Restricted Rights - Use, duplication or
# disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
#

QueryStatus


PURPOSE

	The QueryStatus command queries the status database file and prints to 
standard output any record with fields that matches the fields specified on the
command line.


SYNOPSIS


	 QueryStatus [-a <avalue>] [-b <bvalue>] [-c <cvalue>] [-d <dvalue>]
                     [-e <evalue>] [-f <fvalue>] [-g <gvalue>] [-h <hvalue>]
                     [-i <ivalue>] [-j <jvalue>] [-A] filename

DESCRIPTION

	The QueryStatus command queries the status database file and prints the
records whose fields match those selected on the command line to the standard
output. The QueryStatus command prints only those fields from the matched 
record that were not selected for queries on the command line and whose fields 
are not null. If the '-A' (all) option is chosen then it prints all the fields 
of the matched record. To prevent a field from being displayed,the "don't care"
('*') value can be used for that field on the command line. The QueryStatus 
command always prints the fields in raw format, i.e, fields seperated by '|' 
(bar). If the filename is not specified then it checks for the value of the 
environment variable "STATUS_FILE" and if not set, then it uses the default 
file "$BLDTMP/status.db".
	QueryStatus returns zero on success and one on errors. All errors go to
STDERR.


FLAGS
 

	-a <avalue>      The first field with the string <avalue>

        -b <bvalue>      The second field with the string <bvalue>  

        -c <cvalue>      The third field with the string <cvalue>  

        -d <dvalue>      The fourth field with the string <dvalue>  

        -e <evalue>      The fifth field with the string <evalue>  

        -f <fvalue>      The sixth field with the string <fvalue>  

        -g <gvalue>      The seventh field with the string <gvalue>  

        -h <hvalue>      The eight field with the string <hvalue>
	
	-i <ivalue>      The ninth field with the string <ivalue>  

        -j <jvalue>      The tenth field with the string <jvalue>  

	-A		 Prints all the fields

	filename	 The database file where status information is stored

EXAMPLES

	1. To query for prebuild status with success for the cmvccommit task

		QueryStatus -a prebuild -i cmvccommit -j success

        Output:
	9138|bos320|9138320a|rgjack|Sep 15 11:18
	9138|bos320|9140320|ray|Sep 26 12:42

	2. The same query using the don't care value for the dates field

		QueryStatus -a prebuild -i cmvccommit -j success -d '*'

	Output:
	9138|bos320|9138320a|rgjack
	9138|bos320|9140320|ray

	3. With the same query as above and using the -A option to print all 
	   fields.
		
		QueryStatus -a prebuild -i "cmvccommit" -j success -A

	Output:
	prebuild|9138|bos320|9138320a|rgjack|Sep 15 11:18|||cmvccommit|success
	prebuild|9140|bos320|9140320|ray|Sep 26 12:42|||cmvccommit|success

RELATED INFORMATION

	The following commands
		SetStatus
		CheckStatus
		DeleteStatus
		bldsetstatus
		bldlock

	The following file
		$BLDTMP/status.db


bldenv/bldtools/QueryStatus.pl
#! /usr/bin/perl
# @(#)88	1.9  src/bldenv/bldtools/QueryStatus.pl, bldprocess, bos412, GOLDA411a 1/23/92 14:53:13
#
# COMPONENT_NAME: (BLDPROCESS) BAI Build Process
#
# FUNCTIONS: QueryStatus
#	     process_opt_arg
#	     usage
#	     initialize
#	     get_cmd_line
#
# ORIGINS: 27
#
# IBM CONFIDENTIAL -- (IBM Confidential Restricted when
# combined with the aggregated modules for this product)
#                  SOURCE MATERIALS
# (C) COPYRIGHT International Business Machines Corp. 1991
# All Rights Reserved
#
# US Government Users Restricted Rights - Use, duplication or
# disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
#
# NAME: QueryStatus
#
# FUNCTION: Queries the status of any operation or command in the  database file
#	    and prints the record that matches to the standard output
#
# INPUT: avalue,bvalue,cvalue,dvalue,evalue,filename(global command line
#	 (arguments)
#
# OUTPUT: Standard output
#
# EXECUTION ENVIRONMENT: Build process environment
#

push(@INC,split(/:/,$ENV{"PATH"}));          # Including path 
do 'bldperlconst';
do 'bldstatusfunc';
&initialize('*');
&get_cmd_line;
&get_status_file;
open status_file || die("Can't find the file $status_file\n");
while (<status_file>){

	chop($_);
	($a_db_val,$b_db_val,$c_db_val,$d_db_val,$e_db_val,$f_db_val, $g_db_val,$h_db_val,$i_db_val,$j_db_val) = split(/\|/,$_);

	if((($a_value =~ /^\*$|^don\'t care$/) || ($a_value eq $a_db_val))
	&& (($b_value =~ /^\*$|^don\'t care$/) || ($b_value eq $b_db_val))
	&& (($c_value =~ /^\*$|^don\'t care$/) || ($c_value eq $c_db_val))
        && (($d_value =~ /^\*$|^don\'t care$/) || ($d_value eq $d_db_val))
        && (($e_value =~ /^\*$|^don\'t care$/) || ($e_value eq $e_db_val))
        && (($f_value =~ /^\*$|^don\'t care$/) || ($f_value eq $f_db_val))
        && (($g_value =~ /^\*$|^don\'t care$/) || ($g_value eq $g_db_val))
        && (($h_value =~ /^\*$|^don\'t care$/) || ($h_value eq $h_db_val))
        && (($i_value =~ /^\*$|^don\'t care$/) || ($i_value eq $i_db_val))
        && (($j_value =~ /^\*$|^don\'t care$/) || ($j_value eq $j_db_val))){
		
		local($line) = "";
		if ($A_value){
			$line .= "$a_db_val\|$b_db_val\|$c_db_val\|$d_db_val";
			$line .= "\|$e_db_val\|$f_db_val\|$g_db_val";
			$line .= "\|$h_db_val\|$i_db_val\|$j_db_val\n";
			print "$line";
			next;
		}


		$line .= "$a_db_val" 
			if (($a_db_val ne "") && ($a_value eq "*"));
		$line .= "\|$b_db_val" 
			if (($b_db_val ne "") && ($b_value eq "*"));
		$line .= "\|$c_db_val" 
			if (($c_db_val ne "") && ($c_value eq "*"));
		$line .= "\|$d_db_val" 
			if (($d_db_val ne "") && ($d_value eq "*"));
		$line .= "\|$e_db_val" 
			if (($e_db_val ne "") && ($e_value eq "*"));
		$line .= "\|$f_db_val" 
			if (($f_db_val ne "") && ($f_value eq "*"));
		$line .= "\|$g_db_val" 
			if (($g_db_val ne "") && ($g_value eq "*"));
		$line .= "\|$h_db_val" 
			if (($h_db_val ne "") && ($h_value eq "*"));
		$line .= "\|$i_db_val" 
			if (($i_db_val ne "") && ($i_value eq "*"));
		$line .= "\|$j_db_val" 
			if (($j_db_val ne "") && ($j_value eq "*"));
		$line =~ s/^\|//;
		print "$line\n" if ($line ne "");
	}
}
exit $SUCCESS;
